### PR TITLE
Add support for proxy settings

### PR DESCRIPTION
**Pull request by dhoppe**
This closes #473. @Rudikza Thank you very much for this enhancement.

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -9,6 +9,7 @@
   $install_options        = []
   $sources                = undef
   $puppet_master          = true
+  $proxy                  = ''
```
Comment (bastelfreak) [inline manifests/params.pp]: please default to undef and not empty string

File: manifests/config.pp [hunk 2]
```diff
@@ -62,6 +65,7 @@
   $root_user                                = $r10k::params::root_user,
   $root_group                               = $r10k::params::root_group,
   Stdlib::Absolutepath $puppetconf_path     = $r10k::params::puppetconf_path,
+  String $proxy                             = $r10k::params::proxy,
```
Comment (bastelfreak) [inline manifests/config.pp]:
```suggestion
  Optional[String[1]] $proxy                = $r10k::params::proxy,
```

File: manifests/config.pp [hunk 3]
```diff
@@ -21,6 +21,9 @@
 #   Location of symlink that points to configfile. Default: /etc/r10k.yaml
 # * [*forge_settings*]
 #   Hash containing settings for downloading modules from the Puppet Forge.
+# * [*proxy*]
+#   String containing proxy setting for r10k.yaml.
+#   Default: ''
```
Comment (Dan33l) [inline manifests/config.pp]: `params.pp` says it defaults to `undef`.

Comment (dhoppe) [reply manifests/config.pp]: It does not matter. I will create another pull request, which will refactor a lot of stuff and hopefully we will use data-in-modules and generate a REFERENCE.md.

File: manifests/init.pp [hunk 4]
```diff
@@ -21,6 +21,7 @@
   Hash $forge_settings                                        = $r10k::params::forge_settings,
   Hash $deploy_settings                                       = $r10k::params::deploy_settings,
   $root_user                                                  = $r10k::params::root_user,
+  Optional[String[1]] $proxy                                  = $r10k::params::proxy,
```
Comment (alexjfisher) [inline manifests/init.pp]: `Stdlib::HTTPUrl` instead of `String[1]`?

**Pull request merged**